discovery-provider/src/tasks/celery_app.py

```diff
@@ -9,3 +17,74 @@
 # Celery removes all configured loggers. This setting prevents
 # Celery from overriding the configured logger set in create_celery()
 celery.conf.update({"worker_hijack_root_logger": False})
+
+
+def celery_worker(logger, db, redis, table_name, aggregate_func):
+    # get name of the caller function
+    worker_name = currentframe().f_back.f_code.co_name
+
+    # Define lock acquired boolean
+    have_lock = False
+    # Define redis lock object
+    lock_name = f"index_{table_name}_lock"
+    update_lock = redis.lock(lock_name, timeout=60 * 10)
+    try:
+        # Attempt to acquire lock - do not block if unable to acquire
+        have_lock = update_lock.acquire(blocking=False)
+        if have_lock:
+            start_time = time()
+
+            with db.scoped_session() as session:
+                aggregate_func(session)
+
+            logger.info(
+                f"{worker_name} | Finished updating \
+                {table_name} in: {time()-start_time} sec"
+            )
+        else:
+            logger.info(f"{worker_name} | Failed to acquire {lock_name}")
+    except Exception as e:
+        logger.error(f"{worker_name} | Fatal error in main loop", exc_info=True)
+        raise e
+    finally:
+        if have_lock:
+            update_lock.release()
+
+
+def aggregate_worker(
```

Comment:
we don't want this to be in this file. this is specific to index aggregate user/track/plays. we could move those into a folder with a utils there but would not generalize this much. this is for all 20+ celery tasks